core/common/src/main/java/alluxio/worker/dora/DoraWorker.java

```diff
@@ -37,12 +37,12 @@ public interface DoraWorker extends DataWorker, SessionCleanable {
 
   /**
    * List status from Under File System.
-   * @param path
-   * @param options
-   * @return list of status
+   * @param path the path of a dir or file
+   * @param options the option for listStatus()
+   * @return list of status in the format of a ListStatusPResponse
    * @throws IOException
    */
-  UfsStatus[] listStatus(String path, ListOptions options) throws IOException;
+  ListStatusPResponse listStatus(String path, ListOptions options) throws IOException;
```

Comment:
`ListStatusPResponse` is a data structure defined in gRPC protos.
We should not involve a gRPC data structure in a transport-agnostic component.
We should either return `FileInfo` or `URIStatus` here.

Reply:
Yes, it makes sense.
But, I have more thoughts on this.
If we store an array of FileInfo or URIStatus in Cache, there would be an array of small and standalone objects.
This is very unfriendly to GC in Java.

But if we cache a ListStatusPResponse, it is a large and compact object, which is more memory friendly and easier for GC.
What do you think?

Reply:
Inside `ListStatusPResponse` there is an `ArrayList`, so there's an array of `FileInfo` anyway. Also I don't think java does object inlining at least currently, so an raw ArrayList and an ArrayList in `ListStatusPResponse` is the same in terms of the amount of work GC needs to do to reclaim the objects. 

But yes, creating `ListStatusPResponse` from cached `FileInfo[]` always needs to copy the array, taking up twice the amount of memory. Hopefully `ListStatusPResponse` objects are short-lived, and can be reclaimed very soon. I guess this is the price we have to pay for jumping onboard the gRPC ship.

Reply:
OK. Revert back to the original design.